src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -16,7 +16,20 @@ public static partial class JsonSerializer
             ref Utf8JsonReader reader,
             ref ReadStack state)
         {
-            JsonPropertyInfo jsonPropertyInfo = state.Current.JsonPropertyInfo;
+            JsonPropertyInfo jsonPropertyInfo;
+
+            if (state.Current.IsDictionary())
```

Comment:
nit: should `state.Current` be cached in a local variable? @steveharter, this is a general question across a lot of the usages within the serializer.

Reply:
No it is a struct on a field so a copy will be made.